geomesa-raster/src/main/scala/org/locationtech/geomesa/raster/data/RasterStore.scala

```diff
@@ -17,50 +17,267 @@
 package org.locationtech.geomesa.raster.data
 
 import java.awt.image.BufferedImage
+import java.util.Map.Entry
+import java.util.concurrent.{TimeUnit, Callable}
 
+import com.google.common.cache.CacheBuilder
 import com.google.common.collect.ImmutableSetMultimap
+import org.apache.accumulo.core.client.{TableExistsException, BatchWriterConfig, Connector}
+import org.apache.accumulo.core.data.{Mutation, Value, Key, Range}
+import org.apache.accumulo.core.security.{TablePermission, Authorizations}
 import org.geotools.coverage.grid.GridEnvelope2D
-import org.locationtech.geomesa.raster.AccumuloStoreHelper
+import org.joda.time.DateTime
+import org.locationtech.geomesa.core.index.StrategyHelpers
+import org.locationtech.geomesa.core.iterators.BBOXCombiner._
+import org.locationtech.geomesa.core.security.AuthorizationsProvider
+import org.locationtech.geomesa.core.stats.{RasterQueryStatTransform, RasterQueryStat, StatWriter}
+import org.locationtech.geomesa.core.util.{SelfClosingScanner, SelfClosingBatchScanner}
+import org.locationtech.geomesa.raster._
+import org.locationtech.geomesa.raster.index.RasterIndexSchema
+import org.locationtech.geomesa.raster.util.RasterUtils
 import org.locationtech.geomesa.utils.geohash.BoundingBox
+import org.locationtech.geomesa.utils.stats.{TimingsImpl, MethodProfiling}
 
-/**
- * This class defines basic operations on Raster, including saving/retrieving
- * Raster objects to/from data source.
- *
- * @param rasterOps
- */
-class RasterStore(val rasterOps: RasterOperations) {
+import scala.collection.JavaConversions._
+
+trait RasterOperations extends StrategyHelpers {
+  def getTable(): String
+  def ensureBoundsTableExists(): Unit
+  def createTableStructure(): Unit
+  def getAuths(): Authorizations
+  def getVisibility(): String
+  def getConnector(): Connector
+  def getRasters(rasterQuery: RasterQuery): Iterator[Raster]
+  def getQueryRecords(numRecords: Int): Iterator[String]
+  def putRaster(raster: Raster): Unit
+  def getBounds(): BoundingBox
+  def getAvailableResolutions(): Seq[Double]
+  def getAvailableGeoHashLengths(): Seq[Int]
+  def getAvailabilityMap(): ImmutableSetMultimap[Double, Int]
+  def getGridRange(): GridEnvelope2D
+  def getMosaicedRaster(query: RasterQuery, params: GeoMesaCoverageQueryParams): BufferedImage
+}
+
+class RasterStore(val connector: Connector,
```

Comment:
Do we want to call this an AccumuloRasterStore or something else to indicate the backing store?